Domain/Sia.Shared/Authentication/Certificates/CertificateRetrievalException.cs

```diff
@@ -7,10 +7,19 @@ namespace Sia.Shared.Authentication.Certificates
     public class CertificateRetrievalException : Exception
     {
         private const string ErrorMessage = "Failure when attempting to retrieve certificate";
```

Comment:
There's little reason for this class to exist if the message can be overridden.

What rule requires these constructors to be added?

Reply:
The rules that I applied are called out in the notes for each commit.. in this case the rule is: https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1032-implement-standard-exception-constructors

The rule talks about assumptions the platform makes about exception construction:
> Failure to provide the full set of constructors can make it difficult to correctly handle exceptions. For example, the constructor that has the signature NewException(string, Exception) is used to create exceptions that are caused by other exceptions. Without this constructor you cannot create and throw an instance of your custom exception that contains an inner (nested) exception, which is what managed code should do in such a situation. 

We dont use these constructors for the exception directly in our code and it's reasonable to not have unused code floating around.